apps/prairielearn/src/pages/administratorQuery/administratorQuery.html.ts

```diff
@@ -0,0 +1,333 @@
+            'tablesorter/dist/js/jquery.tablesorter.widgets.min.js',
+          )}></script>
+      </head>
+      <body>
+        ${renderEjs(__filename, "<%- include('../partials/navbar') %>", {
+          ...resLocals,
+          navPage: 'admin',
+          navSubPage: 'queries',
+        })}
+
+        <main id="content" class="container-fluid">
+          <div class="card mb-4">
+            <div class="card-header bg-primary text-white d-flex align-items-center">
+              <span class="font-weight-bold text-monospace">${sqlFilename}</span>
+              <button
+                class="btn btn-xs btn-light ml-2 my-n2"
+                type="button"
+                data-toggle="collapse"
+                data-target="#sql-query"
+                aria-expanded="false"
+                aria-controls="sql-query"
+              >
+                Show SQL<i class="fas fa-caret-down"></i>
+              </button>
+              <span class="ml-3">&mdash;</span>
+              <span class="ml-3">${info.description}</span>
+            </div>
+
+            <div id="sql-query" class="collapse">
+              <pre class="m-0 p-2 bg-light border-bottom"><code class="sql">${unsafeHtml(
+                resLocals.sqlHighlighted,
+              )}</code></pre>
+            </div>
+            <div class="card-body">
+              <form name="run-query-form" method="POST">
+                ${info.params
+                  ? info.params.map((param) => {
+                      return html`
+                        <div class="form-group">
+                          <label for=${`param-${param.name}`}>${param.name}</label>
+                          <input
+                            class="form-control"
+                            type="text"
+                            id=${`param-${param.name}`}
+                            aria-describedby=${`param-${param.name}-help`}
+                            name=${param.name}
+                            autocomplete="off"
+                            ${typeof params != 'undefined' && params != null && params[param.name]
+                              ? html`value=${params[param.name]}`
+                              : param.default != null && param.default !== ''
+                              ? html`value=${param.default}`
+                              : html`value = ''`}
+                          />
+
+                          <small id=${`param-${param.name}-help`} class="form-text text-muted"
+                            >${param.description}</small
+                          >
+                        </div>
+                      `;
+                    })
+                  : null}
+                <input type="hidden" name="__csrf_token" value=${resLocals.__csrf_token} />
+                <button type="submit" class="btn btn-primary">
+                  <i class="fas fa-play"></i>
+                  Run query ${typeof result != 'undefined' ? 'again' : null}
```

Comment:
Result is never undefined. Do you mean null? Same question in other cases where undefined is used.

Reply:
I am not sure why this was using undefined. Changed code to use `has_query_run` instead of `typeof result != 'undefined' in commit 227f6c3.